modules/shared/state.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @typedef {Object} State
+ * @summary simple state object. Can be subclassed
+ * @function updateState
+ * @function getState
+ * @function clearState
+ */
+
+/**
+ * @summary factory to create a simple state object
+ * @returns {State}
+ */
+export default function stateFactory() {
```

Comment:
this does not seem to add much value: why not use vanilla objects?

Reply:
makes the code a bit DRYer. Useful for future video submodules since they will need to track state as well.

Reply:
I disagree somewhat - the abstraction is so thin, I don't think `state.updateState(newState)` is appreciably better than `Object.assign(state, newState)`. It wouldn't be worse either, except that extra code has a cost in the browser. 

But I don't mind it enough to argue any more than this :)